src/main/java/net/minecraftforge/registries/RegistryObject.java

```diff
@@ -102,6 +103,12 @@ public void accept(Predicate<ResourceLocation> pred)
                     RegistryObject.this.value = registry.containsKey(RegistryObject.this.name) ? (T)registry.getValue(RegistryObject.this.name) : null;
             }
         });
+        IForgeRegistry<V> registry = RegistryManager.ACTIVE.getRegistry(baseType);
+        // allow registry to be null, this might be for a custom registry that does not exist yet
```

Comment:
Is this actually needed? I feel like it potentially should fail given custom registries are added/registered before any registry events have fired. That means if we are *after* registry events the registry shouldn't be null unless it is missing/invalid? (See a few lines above this where the registry is null checked.)

Or is this related to it trying to instantly set it and if it is null that means it was created before which currently doesn't have issues as it doesn't run any code at that time?

Reply:
Yes it is needed. Consider the following setup for a custom registry:
```java
private static final DeferredRegister<Custom> CUSTOMS = DeferredRegister.create(Custom.class, MODID);
private static final RegistryObject<Custom> CUSTOM = CUSTOMS.register("test", () -> new Custom());
```
The 2nd line (indirectly) class the `RegistryObject` constructor from a static initializer, while the custom registry is nowhere to be found yet. It is yet to be created by the `DeferredRegister` once `RegistryEvent.NewRegistry` fires.